src/test/java/com/epam/tc/hw3/testcases/Exercise2Parametrized.java

```diff
@@ -0,0 +1,145 @@
+        testDriver.navigate().refresh(); //clean all elements state
+
+        difElementsPage.getAllCheckboxes().get(parameter).click();
+        difElementsPage.getAllCheckboxes().get(parameter).click();
+        softly.assertThat(difElementsPage.getLastLog().getText())
+                .as("Incorrect log text")
+                .endsWith(checkboxName + ": condition changed to " + isSelected);
+
+        softly.assertAll();
+
+        //10. Close Browser - in parent class
+
+    }
+
+    //9. Assert that
+    //• for radio button there is a log row and value is corresponded to the status of radio button
+
+    @Test(dataProvider = "DiffElementsRadiobutton", dataProviderClass = DataProviders.class)
+    public void assertRadiobuttons(int parameter, String radiobuttonName) {
+        SoftAssertions softly = new SoftAssertions();
+        DifElementsPage difElementsPage = new DifElementsPage(testDriver);
+
+        //Checking page is correct
+        softly.assertThat(testDriver.getCurrentUrl())
+                .as("Incorrect page opened")
+                .isEqualTo("https://jdi-testing.github.io/jdi-light/different-elements.html");
+
+        //9. Assert that
+        //• for each checkbox there is an individual log row and value is corresponded to the status of checkbox
+        testDriver.navigate().refresh(); //clean all elements state
+
+        difElementsPage.getAllradiobuttons().get(parameter).click();
+        softly.assertThat(difElementsPage.getLastLog().getText())
+                .as("Incorrect log text")
+                .endsWith("metal: value changed to " + radiobuttonName);
+
+        softly.assertAll();
+
+        //10. Close Browser - in parent class
+
+    }
+
+    //9. Assert that
+    //• for dropdown there is a log row and value is corresponded to the selected value.
+    @Test(dataProvider = "DiffElementsDropdown", dataProviderClass = DataProviders.class)
+    public void assertDropdown(int parameter, String dropdownName) {
+        SoftAssertions softly = new SoftAssertions();
+        Header header = new Header(testDriver);
+
+        WebDriverWait wait = new WebDriverWait(testDriver, Duration.ofSeconds(10));
+
+        //Checking page is correct
+        softly.assertThat(testDriver.getCurrentUrl())
+                .as("Incorrect page opened")
+                .isEqualTo("https://jdi-testing.github.io/jdi-light/different-elements.html");
+
+        DifElementsPage difElementsPage = new DifElementsPage(testDriver);
+
+        //9. Assert that
+        //• for each checkbox there is an individual log row and value is corresponded to the status of checkbox
+
+        difElementsPage.getAlldropDown().get(parameter).click();
+        softly.assertThat(difElementsPage.getLastLog().getText())
+                .as("Incorrect log text")
+                .endsWith("Colors: value changed to " + dropdownName);
+        softly.assertAll();
+
+        //10. Close Browser - in parent class
+
+    }
+}
```

Comment:
What is it? I don't see it in hw3 doc file. I think this part should be in Exercise2

Reply:
This was added for using DataProviders. I misunderstood one of the requirements. Redesigned into Exercise2 class part.